OSDOCS#13403: OSDK 1.38.0 migration docs for OCP 4.18 GA (#88859)
openshift/openshift-docs: modules/osdk-updating-1361-to-138.adoc

// Module included in the following assemblies:
//
// * operators/operator_sdk/golang/osdk-golang-updating-projects.adoc
// * operators/operator_sdk/ansible/osdk-ansible-updating-projects.adoc
// * operators/operator_sdk/helm/osdk-helm-updating-projects.adoc

ifeval::["{context}" == "osdk-golang-updating-projects"]
:golang:
:type: Go
endif::[]
ifeval::["{context}" == "osdk-ansible-updating-projects"]
:ansible:
:type: Ansible
endif::[]
ifeval::["{context}" == "osdk-helm-updating-projects"]
:helm:
:type: Helm
endif::[]

:_mod-docs-content-type: PROCEDURE
[id="osdk-upgrading-projects_{context}"]
= Updating {type}-based Operator projects for Operator SDK {osdk_ver}

The following procedure updates an existing {type}-based Operator project for compatibility with {osdk_ver}.

.Prerequisites

* Operator SDK {osdk_ver} installed
* An Operator project created or maintained with Operator SDK {osdk_ver_n1}

.Procedure

// The following few steps should be retained/updated for each new migration procedure, as they're just bumping the OSDK version for each language type.

> . Edit your Operator project's Makefile to update the Operator SDK version to {osdk_ver}, as shown in the following example:

Reviewer: ```suggestion
. Edit the Makefile of your Operator project to update the Operator SDK version to {osdk_ver}, as shown in the following example:
```

+
.Example Makefile
[source,make,subs="attributes+"]
----
# Set the Operator SDK version to use. By default, what is installed on the system is used.
# This is useful for CI or a project to utilize a specific version of the operator-sdk toolkit.
OPERATOR_SDK_VERSION ?= v{osdk_ver} <1>
----
<1> Change the version from `{osdk_ver_n1}` to `{osdk_ver}`.
ifdef::helm[]

> . Edit your Operator's Dockerfile to update the `ose-helm-rhel9-operator` image tag to `{product-version}`, as shown in the following example:

Reviewer: ```suggestion
. Edit the Dockerfile of your Operator to update the `ose-helm-rhel9-operator` image tag to `{product-version}`, as shown in the following example:
```

+
.Example Dockerfile
[source,docker,subs="attributes+"]
----
FROM registry.redhat.io/openshift4/ose-helm-rhel9-operator:v{product-version}
----
endif::helm[]

ifdef::ansible[]

> . Edit your Operator's Dockerfile to update the `ose-ansible-operator` image tag to `{product-version}`, as shown in the following example:

Reviewer: ```suggestion
. Edit the Dockerfile of your Operator to update the `ose-ansible-operator` image tag to `{product-version}`, as shown in the following example:
```

+
.Example Dockerfile
[source,docker,subs="attributes+"]
----
FROM registry.redhat.io/openshift4/ose-ansible-operator:v{product-version}
----
endif::ansible[]

. You must upgrade the Kubernetes versions in your Operator project to use 1.30 and Kubebuilder v4.
+
[TIP]
====
This update include complex scaffolding changes due to the removal of link:https://github.com/brancz/kube-rbac-proxy[kube-rbac-proxy]. If these migrations become difficult to follow, scaffold a new sample project for comparison.
====

ifdef::helm,ansible[]
.. Update the Kustomize version in your Makefile by making the following changes:
+
[source,diff]
----

> - curl -sSLo - https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize/v5.3.0/kustomize_v5.3.0_$(OS)_$(ARCH).tar.gz | \

Author: This can be ignored.
